media-gfx/zbar/zbar-0.23.ebuild

```diff
@@ -0,0 +1,210 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit autotools flag-o-matic java-pkg-opt-2 multilib-minimal python-single-r1 virtualx
+
+DESCRIPTION="Library and tools for reading barcodes from images or video"
+HOMEPAGE="https://github.com/mchehab/zbar"
+SRC_URI="https://linuxtv.org/downloads/${PN}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+
+IUSE="dbus graphicsmagick gtk +imagemagick introspection java jpeg python qt5 static-libs test +threads v4l X xv"
+REQUIRED_USE="
+	introspection? ( gtk )
+	jpeg? ( imagemagick )
```

Comment:
this is a new condition since 0.20 - is it actually true? Because then I'm wondering why `jpeg? (virtual/jpeg)` still exists in the ebuild (and whether imagemagick isn't simply replacing that in the new version).

Reply:
Yes.

![Selection_134](https://user-images.githubusercontent.com/8144644/73650937-63b2fc80-4694-11ea-850c-df9ba82143e4.jpg)

it also [required](https://github.com/mchehab/zbar/blob/1d51925e6cc9151f4a73781989f21fcd7b57ef32/zbar/Makefile.am#L107) `virtual/jpeg` without ImageMagick and I get zbar configuring errors when I remove `virtual/jpeg` (media-libs/libjpeg-turbo)

https://github.com/mchehab/zbar/blob/1d51925e6cc9151f4a73781989f21fcd7b57ef32/zbar/jpeg.c#L25

Reply:
okay, but doesn't that mean you have the condition backwards?

Reply:
>it also required virtual/jpeg without ImageMagick and I get zbar configuring errors when I remove jpeg using emerge -C media-libs/libjpeg-turbo && USE="jpeg" ebuild zbar-0.23.ebuild configure command

>https://github.com/mchehab/zbar/blob/1d51925e6cc9151f4a73781989f21fcd7b57ef32/zbar/jpeg.c#L25

the jpeg library needs for grabbing QR-codes from the video stream (`v4l` use flag for enabling) and convert that to another format. Unfortunately, I'm not a developer and I need more time for resolving/understanding this in details

API docs for legacy 0.10 version: https://www.ccoderun.ca/programming/doxygen/zbar/video_8c.html#ab05beba9fcb9631de884b7f4a829eb59 and https://www.ccoderun.ca/programming/doxygen/zbar/jpeg_8c.html#ace97980383111502ac0c1bc31fee3459

>okay, but doesn't that mean you have the condition backwards?

`jpeg? ( imagemagick )` — is not necessary in this case I think
I need to remove that rule